spec/controllers/users/phones_controller_spec.rb

```diff
@@ -16,6 +16,30 @@
       get :add
 
       expect(response).to render_template(:add)
+      expect(response.request.flash[:alert]).to be_nil
+    end
+
+    it 'displays error if phone number exceeds limit' do
+      user.phone_configurations.create(encrypted_phone: '4105555551')
+      user.phone_configurations.create(encrypted_phone: '4105555552')
+      user.phone_configurations.create(encrypted_phone: '4105555553')
+      user.phone_configurations.create(encrypted_phone: '4105555554')
+      controller.request.headers.merge({ HTTP_REFERER: account_url })
+
+      get :add
+      expect(response).to redirect_to(account_url(anchor: 'phones'))
+      expect(response.request.flash[:phone_error]).to_not be_nil
+    end
+
+    it 'renders the #phone anchor when it exceeds limit' do
+      user.phone_configurations.create(encrypted_phone: '4105555551')
+      user.phone_configurations.create(encrypted_phone: '4105555552')
+      user.phone_configurations.create(encrypted_phone: '4105555553')
+      user.phone_configurations.create(encrypted_phone: '4105555554')
+      controller.request.headers.merge({ HTTP_REFERER: account_url })
```

Comment:
Should one of these test cases be testing the referer matching? Right now they seem to be testing mostly the same behavior. I could imagine one using the `account_two_factor_authentication_url` as the referer to test it redirects to the right place?

Reply:
One is testing that its redirecting and has an error. 
The other is testing to make sure the anchor tag #phone is in the url . 
Not sure what you mean?

Reply:
> its redirecting [...] the anchor tag #phone is in the url

To me, these seem like the same thing, i.e. the redirect location has the anchor tag.

But the bigger thing for me was test coverage for `request.referer.match`, since we're only testing the one branch of the condition (the fallback).

Reply:
ah gotcha. I refactored it too.  I rather keep the one where we are checking the anchor tag. Since it can get accidentally stripped. As we saw earlier with the other way we were trying to redirect.